tests/publiccloud/migration.pm

```diff
@@ -0,0 +1,56 @@
+# SUSE's openQA tests
+#
+# Copyright 2016-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Basic DMS test
+#
+#   This test does the following
+#    - Installs SLE15-Migration and sle15-activation
+#    - Reboot the system
+#    - Check the system version is the expected one
+#
+# Maintainer: Jesus Bermudez Velazquez <jesus.bv@suse.com>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use publiccloud::utils;
+
+our $target_version = get_required_var('TARGET_VERSION');
+
+sub run {
+    my ($self, $args) = @_;
+    $self->select_serial_terminal;
+    my $provider = $self->provider_factory();
+    my $instance = $provider->create_instance();
+    $instance->wait_for_guestregister();
+
+    record_info('INFO', $target_version);
+    record_info('INFO', get_var('TARGET_VERSION'));
+
+    sleep 90;    # wait for a bit for zypper to be available
```

Comment:
Is this really needed? The general rule is to not blindly wait.

Reply:
Yes, zypper is usually busy once an instance starts, so better wait and be sure that zypper is available (we faced that problem already doing this test the first time)

Reply:
Why don't you instead detect if zypper is already available? There is `utils::script_retry` for it.